cgmes/cgmes-completion/src/main/java/com/powsybl/cgmes/completion/CreateMissingContainersPreProcessor.java

```diff
@@ -0,0 +1,216 @@
+        // TODO(Luma) should we ensure that context CIM version uses the same version of the input files ?
+        CgmesExportContext context = new CgmesExportContext(network);
+        try (ZipOutputStream zout = new ZipOutputStream(Files.newOutputStream(fixesFile))) {
+            zout.putNextEntry(new ZipEntry(basename + "_EQ.xml"));
+            XMLStreamWriter writer = XmlUtil.initializeWriter(true, "    ", zout);
+            writeHeader(writer, context);
+            RegionContainers regionContainers = writeRegionContainers(writer, context);
+            for (String missingVoltageLevel : missingVoltageLevels) {
+                writeMissingVoltageLevel(missingVoltageLevel, writer, context, regionContainers);
+            }
+            writer.writeEndDocument();
+            zout.closeEntry();
+        } catch (IOException | XMLStreamException x) {
+            throw new PowsyblException("Building file containing fixes for missing data", x);
+        }
+    }
+
+    private static RegionContainers writeRegionContainers(XMLStreamWriter writer, CgmesExportContext context) throws XMLStreamException {
+        String cimNamespace = context.getCim().getNamespace();
+
+        // TODO(Luma) An alternative would be to make public the method writeFictitiousSubstationFor()
+        // from EquipmentExport and use it here.
+        // We could group all missing voltage levels in the same (fictitious) substation
+        RegionContainers regionContainers = new RegionContainers();
+        regionContainers.subGeographicalRegionId = CgmesExportUtil.getUniqueId();
+        String subGeographicalRegionName = "SGR fix for missing data";
+        regionContainers.geographicalRegionId = CgmesExportUtil.getUniqueId();
+        String geographicalRegionName = "GR fix for missing data";
+        SubGeographicalRegionEq.write(regionContainers.subGeographicalRegionId, subGeographicalRegionName, regionContainers.geographicalRegionId, cimNamespace, writer, context);
+        GeographicalRegionEq.write(regionContainers.geographicalRegionId, geographicalRegionName, cimNamespace, writer, context);
+        return regionContainers;
+    }
+
+    private static void writeMissingVoltageLevel(String voltageLevelId, XMLStreamWriter writer, CgmesExportContext context, RegionContainers regionContainers) throws XMLStreamException {
+        String cimNamespace = context.getCim().getNamespace();
+
+        // TODO(Luma) In a first approach,
+        // we do not have additional information about the voltage level,
+        // we create a different substation and base voltage for every missing voltage level
+        String voltageLevelName = voltageLevelId + " VL";
+        String substationId = CgmesExportUtil.getUniqueId();
+        String substationName = voltageLevelId + "SUB for missing VL " + voltageLevelId;
+        String baseVoltageId = CgmesExportUtil.getUniqueId();
+        // An arbitrary, abnormal value
+        double nominalV = 10000.0;
```

Comment:
Maybe put it as parametrisable? At minimum, it should be a constant to be clear, not hidden in the code

Reply:
It is not very useful to parametrise it, the missing voltage levels may correspond to different points of the network. I added a constant.